File: scripts/birth_abortion_relationship.R
library(dplyr)
library(ggplot2)
library(ggrepel)

# Load the data
abortion_diffs <- read.csv('data/abortion_diffs.csv')
birth_diffs <- read.csv('data/birth_diffs.csv')

# Calculate actual counts from rates
abortion_diffs$AbortionCount <- abortion_diffs$AbortionsPerThousandWomen * abortion_diffs$Population / 1000
birth_diffs$BirthCount <- birth_diffs$BirthsPerThousandWomen * birth_diffs$Population / 1000

# Merge the data
merged_data <- full_join(
  abortion_diffs %>% select(State, Year, Month, AbortionCount, TreatedOriginal),
  birth_diffs %>% select(State, Year, Month, BirthCount),
  by = c("State", "Year", "Month")
)

# Create the plot
plot <- ggplot(merged_data, aes(x = AbortionCount, y = BirthCount, color = TreatedOriginal)) +
  geom_point(alpha = 0.6, size = 2) +
  geom_smooth(method = "lm", se = TRUE, alpha = 0.2) +
  scale_color_manual(values = c("Control" = "blue", "Neighbor" = "darkgreen", "Ban" = "red")) +
  theme_bw() +
  labs(
    x = "Monthly Abortion Count",
    y = "Monthly Birth Count",
    color = "State Type",
    title = "Relationship Between Monthly Abortion and Birth Counts",
    subtitle = "By State Type (2022-2023)"
  ) +
  theme(
    text = element_text(size = 12),
    plot.title = element_text(hjust = 0.5, size = 16),
    plot.subtitle = element_text(hjust = 0.5, size = 12),
    legend.position = "bottom"
  )

# Save the plot
ggsave('figures/birth_abortion_relationship.png', plot, width = 10, height = 8, dpi = 300)

# Print summary statistics
cat("\nSummary Statistics:\n")
cat("Correlation between abortion and birth counts:", 
    cor(merged_data$AbortionCount, merged_data$BirthCount, use = "complete.obs"), "\n")

# Print regression results
model <- lm(BirthCount ~ AbortionCount + TreatedOriginal, data = merged_data)
cat("\nRegression Results:\n")
print(summary(model)) 

File: scripts/birth_abortion_relationship.py
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.linear_model import LinearRegression
import numpy as np

# Set the style for the plots
plt.style.use('seaborn')
sns.set_palette(['blue', 'darkgreen', 'red'])

# Load the data
abortion_diffs = pd.read_csv('data/abortion_diffs.csv')
birth_diffs = pd.read_csv('data/birth_diffs.csv')

# Calculate actual counts from rates
abortion_diffs['AbortionCount'] = abortion_diffs['AbortionsPerThousandWomen'] * abortion_diffs['Population'] / 1000
birth_diffs['BirthCount'] = birth_diffs['BirthsPerThousandWomen'] * birth_diffs['Population'] / 1000

# Merge the data
merged_data = pd.merge(
    abortion_diffs[['State', 'Year', 'Month', 'AbortionCount', 'TreatedOriginal']],
    birth_diffs[['State', 'Year', 'Month', 'BirthCount']],
    on=['State', 'Year', 'Month'],
    how='outer'
)

# Create the plot
plt.figure(figsize=(10, 8))
sns.scatterplot(
    data=merged_data,
    x='AbortionCount',
    y='BirthCount',
    hue='TreatedOriginal',
    alpha=0.6,
    s=100
)

# Add regression lines for each group
for group in merged_data['TreatedOriginal'].unique():
    group_data = merged_data[merged_data['TreatedOriginal'] == group]
    if len(group_data) > 1:  # Only plot regression if we have enough points
        model = LinearRegression()
        X = group_data[['AbortionCount']]
        y = group_data['BirthCount']
        model.fit(X, y)
        x_range = np.linspace(X.min(), X.max(), 100)
        y_pred = model.predict(x_range.reshape(-1, 1))
        plt.plot(x_range, y_pred, label=f'{group} trend', alpha=0.5)

plt.title('Relationship Between Monthly Abortion and Birth Counts\nBy State Type (2022-2023)')
plt.xlabel('Monthly Abortion Count')
plt.ylabel('Monthly Birth Count')
plt.legend(title='State Type')

# Save the plot
plt.tight_layout()
plt.savefig('figures/birth_abortion_relationship.png', dpi=300, bbox_inches='tight')

# Print summary statistics
print("\nSummary Statistics:")
print(f"Correlation between abortion and birth counts: {merged_data['AbortionCount'].corr(merged_data['BirthCount']):.3f}")

# Print regression results
print("\nRegression Results:")
# Create dummy variables for TreatedOriginal
dummies = pd.get_dummies(merged_data['TreatedOriginal'], drop_first=True)
X = pd.concat([merged_data[['AbortionCount']], dummies], axis=1)
y = merged_data['BirthCount']

model = LinearRegression()
model.fit(X, y)

# Print coefficients
print("\nCoefficients:")
for feature, coef in zip(X.columns, model.coef_):
    print(f"{feature}: {coef:.3f}")
print(f"Intercept: {model.intercept_:.3f}")

# Print R-squared
print(f"\nR-squared: {model.score(X, y):.3f}") 